File: Programmers/Lv.1/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    matched = 0
    possible = 0
    for num in lottos:
        if num in win_nums:
            matched += 1
        elif num == 0:
            possible += 1
    
    # 최고
    if matched == 0 and possible == 0: 
        answer.append(6)
    else:
        answer.append(7 - (matched + possible))  
    # 최저
    if matched == 0 or possible == 6:
        answer.append(6)
    else:
        answer.append(7 - matched)
        
    return answer

File: Programmers/Lv.1/로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = {6, 6};
        boolean[] nums = new boolean[46];
        
        for (int num: win_nums) {
            nums[num] = true;
        }
        
        int common = 0;
        int zero = 0;
        for (int num: lottos) {
            if (num == 0)
                zero += 1;
            else if (nums[num]) {
                common += 1;
            }
        }
        
        if (common+zero > 1) 
            answer[0] = 7-(common+zero);
        if (common > 1)
            answer[1] = 7-common;
        
        return answer;
    }
}